backend/src/src-wsimport/src/main/java/org/eclipse/sw360/wsimport/rest/WsRestClient.java

```diff
@@ -53,17 +55,23 @@ private HttpClient getConfiguredHttpClient() {
 
     private HttpResponse getWsConnection(String input, HttpClient client, String serverUrl) throws IOException{
         HttpPost request = new HttpPost(serverUrl);
-        request.addHeader(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON);
-        StringEntity in = new StringEntity(input, ContentType.create(APPLICATION_JSON));
-        request.setEntity(in);
+        request.addHeader(HttpHeaders.CONTENT_TYPE, TranslationConstants.APPLICATION_JSON);
+        StringEntity stringEntity = new StringEntity(input, ContentType.create(TranslationConstants.APPLICATION_JSON));
+        request.setEntity(stringEntity);
         return client.execute(request);
     }
 
-    String getData(String requestString, String token, WsTokenType type, TokenCredentials tokenCredentials) throws IOException {
+    String getData(String requestString, String token, WsTokenType type, TokenCredentials tokenCredentials) throws IOException, HttpException {
         LOGGER.info("Making REST call to " + tokenCredentials.getServerUrl() + " with request: " + requestString + " and token: " + token + " and userKey: " + tokenCredentials.getUserKey());
         String input = generateRequestBody(requestString, tokenCredentials.getUserKey(), type, token);
         HttpClient httpClient = getConfiguredHttpClient();
         HttpResponse response = getWsConnection(input, httpClient, tokenCredentials.getServerUrl());
-        return IOUtils.toString(response.getEntity().getContent(), "UTF-8");
+        int statusCode = response.getStatusLine().getStatusCode();
+        if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+            return IOUtils.toString(response.getEntity().getContent(), "UTF-8");
+        } else {
+            LOGGER.info("Request unsuccessful: " + response.getStatusLine().getReasonPhrase());
+            throw new HttpException();
```

Comment:
maybe add also the message to the exception? Especially since it is logged later.

Reply:
Good point. Added log message